src/main.rs

```diff
@@ -43,109 +58,193 @@ impl AppWeak {
 }
 
 struct AppInner {
+    application: gtk::Application,
     main_window: Option<gtk::ApplicationWindow>,
+    pipeline: Option<gst::Pipeline>,
+    error: Option<Box<dyn error::Error>>,
 }
 
-fn build_menu(_app: &App, application: &gtk::Application) {
-    let menu = gio::Menu::new();
-
-    menu.append("Quit", "app.quit");
-
-    let about_menu = gio::MenuItem::new("About", "app.about");
-    menu.append_item(&about_menu);
-
-    // Connect the about menu entry click event.
+fn build_actions(_app: &App, application: &gtk::Application) {
+    // Create app.about action for the about dialog
+    //
+    // This can be activated from anywhere where we have access
+    // to the application, not just the main window
     let about = gio::SimpleAction::new("about", None);
+
+    // When activated, show an about dialog
     let weak_application = application.downgrade();
-    about.connect_activate(move |_, _| {
+    about.connect_activate(move |_action, _parameter| {
         let application = upgrade_weak!(weak_application);
+
         let p = gtk::AboutDialog::new();
+
         p.set_authors(&["Sebastian Dröge", "Guillaume Gomez"]);
-        p.set_website_label(Some("github repository"));
-        p.set_website(Some(
-            "https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop",
-        ));
-        p.set_comments(Some("A webcam viewer written with gtk-rs and gstreamer-rs"));
-        p.set_copyright(Some("This is under MIT license"));
+        p.set_website_label("github repository");
+        p.set_website("https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop");
+        p.set_comments("A webcam viewer written with gtk-rs and gstreamer-rs");
+        p.set_copyright("This is under MIT license");
         if let Some(window) = application.get_active_window() {
-            p.set_transient_for(Some(&window));
+            p.set_transient_for(&window);
         }
         p.set_modal(true);
-        p.set_program_name("RustFest GTK+GStreamer");
+        p.set_program_name("RustFest 2018 GTK+ & GStreamer WebCam Viewer");
+
+        // When any response on the dialog happens, we simply destroy it.
+        //
+        // We don't have any custom buttons added so this will only ever
+        // handle the close button, otherwise we could distinguish the
+        // buttons by the response
+        p.connect_response(|dialog, _response| {
```

Comment:
You sure about this? What happens when you click on other buttons than "close"?

Reply:
Which other buttons?

Reply:
There are three buttons on an about dialog, no?

Reply:
The close button, and the credits/about "tab" buttons. Only the first causes a response, and any other buttons one would add

Reply:
Good to know. :)